internal/plugins/src/main/kotlin/org/jsonschema2dataclass/internal/plugin/publishing/javadoc.kt

```diff
@@ -0,0 +1,62 @@
+package org.jsonschema2dataclass.internal.plugin.publishing
+
+import basePluginExtension
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.attributes.DocsType
+import org.gradle.api.component.AdhocComponentWithVariants
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.plugins.internal.JvmPluginsHelper
+import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.support.serviceOf
+import org.jetbrains.dokka.gradle.DokkaTask
+
+private const val DOKKA_TASK_NAME: String = "dokkaJavadoc"
+
+fun applyDokka(project: Project, javaPluginExtension: JavaPluginExtension) {
+    project.plugins.apply("org.jetbrains.dokka")
+
+    setupDoc(project)
+    javaPluginExtension.withDokkaJar(
+        project,
+        project.tasks.named(
+            DOKKA_TASK_NAME,
+        ),
+    )
+}
+
+private fun setupDoc(project: Project) {
+    project.tasks.named<DokkaTask>(DOKKA_TASK_NAME) {
+        moduleName.set(this.project.basePluginExtension.archivesName)
+        // TODO https://github.com/Kotlin/dokka/issues/1894
+        dokkaSourceSets.configureEach {
+            reportUndocumented.set(false)
+        }
+    }
+}
+
+/**
+ * Attaches Dokka as "JavaDoc" provider
+ * @see [org.gradle.api.plugins.JavaPluginExtension.withJavadocJar]
+ *     implementation for details
+ */
+private fun JavaPluginExtension.withDokkaJar(project: Project, artifactTask: TaskProvider<out Task>) {
+    JvmPluginsHelper.configureDocumentationVariantWithArtifact(
+        JavaPlugin.JAVADOC_ELEMENTS_CONFIGURATION_NAME,
+        null,
+        DocsType.JAVADOC,
+        emptyList(),
+        sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME).javadocJarTaskName,
+        artifactTask,
+        project.components["java"] as AdhocComponentWithVariants,
+        project.configurations,
+        project.tasks,
+        project.objects,
+        project.serviceOf(),
+        project.serviceOf(),
+    )
+}
```

Comment:
you'll be soon interested in https://github.com/TWiStErRob/net.twisterrob.gradle/pull/440/files#diff-4636d22ced2cb69af81dc15fd16310e76ef3b23739ac405e57d0a2f105778538

Reply:
Thank you, it helped a lot

Reply:
actually I found a bug in it

Reply:
https://github.com/TWiStErRob/net.twisterrob.gradle/pull/466/commits/c37c20bae5e7fb261c3ca25f1811dffeef385068

also I recently moved this repo to a new format, see the PR this commit is in ^.